language/move-lang/src/parser/syntax.rs

```diff
@@ -1,4 +1,1412 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-include!(concat!(env!("OUT_DIR"), "/parser/syntax.rs"));
+use codespan::{ByteIndex, Span};
+use std::str::FromStr;
+
+use crate::parser::ast::*;
+use crate::parser::lexer::*;
+use crate::shared::*;
+
+// In the informal grammar comments in this file, Comma<T> is shorthand for:
+//      (<T> ",")* <T>?
+// Note that this allows an optional trailing comma.
+
+// FIXME: The following simplified version of ParseError copied from
+// lalrpop-util should be replaced.
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ParseError<L, E> {
+    InvalidToken { location: L },
+    User { location: L, error: E },
+}
+
+fn make_loc(file: &'static str, start: usize, end: usize) -> Loc {
+    Loc::new(
+        file,
+        Span::new(ByteIndex(start as u32), ByteIndex(end as u32)),
+    )
+}
+
+fn spanned<T>(file: &'static str, start: usize, end: usize, value: T) -> Spanned<T> {
+    Spanned {
+        loc: make_loc(file, start, end),
+        value,
+    }
+}
+
+fn consume_token<'input>(
+    tokens: &mut Lexer<'input>,
+    tok: Tok,
+) -> Result<(), ParseError<usize, &'static str>> {
+    if tokens.peek() != tok {
+        return Err(ParseError::InvalidToken {
+            location: tokens.start_loc(),
+        });
+    }
+    tokens.advance()?;
+    Ok(())
+}
+
+// If the next token is the specified kind, consume it and return
+// its source location.
+
+fn consume_optional_token_with_loc<'input>(
+    tokens: &mut Lexer<'input>,
+    tok: Tok,
+) -> Result<Option<Loc>, ParseError<usize, &'static str>> {
+    if tokens.peek() == tok {
+        let start_loc = tokens.start_loc();
+        tokens.advance()?;
+        let end_loc = tokens.previous_end_loc();
+        Ok(Some(make_loc(tokens.file_name(), start_loc, end_loc)))
+    } else {
+        Ok(None)
+    }
+}
+
+// Parse a name:
+//      Name = <NameValue>
+//      MName = <MNameValue>
+//
+// This works for either Tok::NameValue or Tok::MNameValue
+// tokens. The caller is responsible for checking the token kind.
+
```

Comment:
nit, why the space between function and the comment?

Reply:
I was trying to make the comments with the grammar stand out, and then figured it would be good to be consistent. Would you prefer to avoid those blank lines?

Reply:
I prefer large section headers (https://github.com/libra/libra/blob/master/language/move-lang/src/typing/ast.rs#L19) 

And then comments on each function (no space)
It might also be worthwhile using `///` for rustdocs

Reply:
OK, well, my goal was consistency, so I should try to match your style. I'll kill the blank lines and see if there are large sections that merit headers. There's only one public function in that file (parse_file_string) so I can add rustdoc comments for that one.

Reply:
I made those changes. What do you think, @tnowacki ?